modules/raycast/raycast_occlusion_cull.cpp

```diff
@@ -80,23 +80,22 @@ void RaycastOcclusionCull::RaycastHZBuffer::resize(const Size2i &p_size) {
 	memset(camera_ray_masks.ptr(), ~0, camera_rays_tile_count * TILE_RAYS * sizeof(uint32_t));
 }
 
-void RaycastOcclusionCull::RaycastHZBuffer::update_camera_rays(const Transform3D &p_cam_transform, const Projection &p_cam_projection, bool p_cam_orthogonal) {
+void RaycastOcclusionCull::RaycastHZBuffer::update_camera_rays(const Transform3D &p_cam_transform, const Vector3 &p_near_bottom_left, const Vector2 p_near_extents, float p_z_far, bool p_cam_orthogonal) {
```

Comment:
I did wonder if `p_z_far` should be `real` but it looks like there are already assumptions for 32 bit in the `CameraRayThreadData`.

Reply:
Presumably it should it be `const Vector2 &p_near_extents` if it is const?

Reply:
Right. That being said I always wondered if it wouldn't allow more compiler optimization to pass such small structs (2 `real`s ...) by value

Reply:
On your first point, I'll make it `real`. It doesn't cost much

Reply:
Generally in Godot we go with `const &` for anything over 64 bit.

In this case I'd hazard a guess the compiler won't care either way in release, as you've marked it as `const`.